packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -579,48 +579,82 @@ impl<'m> RunnerImpl<'m> {
         // Arrow and pyarrow.
         let target_len = len.unwrap_or(data.len);
 
-        // TODO: Extra copies when generating buffers, because JS Arrow doesn't align things
-        //   properly. (Due to which buffer capacities are currently unused.)
+        // TODO: We currently copy the buffers because the JavsScript representation of arrays does
```

Comment:
```suggestion
        // TODO: We currently copy the buffers because the JavaScript representation of arrays does
```

Can you say how we copy the buffers. As in what's causing the copy (I believe you said it's `Buffer::from_slice_ref`?